server/src/auth/middleware/authMiddleware.ts

```diff
@@ -0,0 +1,21 @@
+import { RequestHandler } from 'express';
+import jwt from 'jsonwebtoken';
+
+import { prisma } from '~/common/prisma';
+
+export const authMiddleware: RequestHandler = async (req, res, next) => {
+  try {
+    const token = req.get('accessToken');
+    if (!token) throw new Error();
+
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    const decodedToken = jwt.verify(token, process.env.SECRET_TOKEN!);
+    const user = await prisma.user.findUnique({ where: { id: (decodedToken as { data: string }).data } });
```

Comment:
Nie twórz takich inlinowych typów, zrób proszę typ dla TokenContent (nawet wrzuciłbym go w auth/typings). Rzutowanie na ten typ przesunąłbym do linii gdzie tworzysz zmienną decodedToken.

Tak przy okazji, to zmieniłbym nazwę właściwości w obiekcie z data na np: userId

Reply:
zrobione